tools/test.py

```diff
@@ -15,19 +15,31 @@
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
+authstring = os.environ.get('USERPASS')
+authstring = ("" if authstring is None else authstring + "@")
+
 class TestMiniKeyValue(unittest.TestCase):
   maxDiff = None
   
   def get_fresh_key(self):
-    return b"http://localhost:3000/swag-" + binascii.hexlify(os.urandom(10))
+    return b"http://"+str.encode(authstring)+b"localhost:3000/swag-" + binascii.hexlify(os.urandom(10))
+
+  # handle 302 manually https://github.com/psf/requests/issues/2949
+  def handle_302(self, key, headers={}, req_type='GET'):
+    if req_type == 'GET': request_funcion = requests.get
+    if req_type == 'HEAD': request_funcion = requests.head
+    r = request_funcion(key, headers=headers, allow_redirects=False)
```

Comment:
request_funcion is misspelled. And I don't think the name of this function should be handle_302